server/src/test/java/org/apache/cassandra/sidecar/acl/authentication/MutualTLSAuthenticationHandlerTest.java

```diff
@@ -218,6 +233,47 @@ void testCertificateWithoutIdentity(VertxTestContext testContext) throws Excepti
               }));
     }
 
+    @Test
+    void testAuthHandlerChaining() throws Exception
+    {
+        Map<String, String> params = new HashMap<>()
+        {
+            {
+                put("certificate_validator", "io.vertx.ext.auth.mtls.impl.CertificateValidatorImpl");
+                put("certificate_identity_extractor", "org.apache.cassandra.sidecar.acl.authentication.CassandraIdentityExtractor");
+            }
+        };
+        ChainAuthHandlerImpl chainAuthHandlerSucceeds = (ChainAuthHandlerImpl) ChainAuthHandler.any();
+        MutualTlsAuthenticationHandlerFactory mTLSAuthFactory = injector.getInstance(MutualTlsAuthenticationHandlerFactory.class);
+        SidecarConfiguration config = injector.getInstance(SidecarConfiguration.class);
+        chainAuthHandlerSucceeds.add(mTLSAuthFactory.create(vertx, config.accessControlConfiguration(), params));
+        chainAuthHandlerSucceeds.add(new TestAuthHandler(new NoOpAuthentication(), true));
+
+        RoutingContext mockContext = mock(RoutingContext.class);
+        HttpServerRequest mockServerRequest = mock(HttpServerRequest.class);
+        HttpConnection mockHttpConnection = mock(HttpConnection.class);
+        when(mockServerRequest.connection()).thenReturn(mockHttpConnection);
+        when(mockContext.request()).thenReturn(mockServerRequest);
+
+        ChainAuthHandlerImpl chainAuthHandlerFails = (ChainAuthHandlerImpl) ChainAuthHandler.any();
+        chainAuthHandlerFails.add(mTLSAuthFactory.create(vertx, config.accessControlConfiguration(), params));
+        chainAuthHandlerFails.add(new TestAuthHandler(new NoOpAuthentication(), false));
+
+        CountDownLatch waitForResponse = new CountDownLatch(2);
+        // mTLS handler will fail without SSL, results succeeds with TestAuthHandler
+        chainAuthHandlerSucceeds.authenticate(mockContext, res -> {
+            assertThat(res.succeeded()).isTrue();
+            waitForResponse.countDown();
+        });
+
+        // response fails when TestAuthHandler fails too
+        chainAuthHandlerFails.authenticate(mockContext, res -> {
+            assertThat(res.failed()).isFalse();
+            waitForResponse.countDown();
+        });
+        waitForResponse.await(1, TimeUnit.MINUTES);
```

Comment:
```suggestion
        assertThat(waitForResponse.await(1, TimeUnit.MINUTES)).isTrue();
```
Otherwise, it just holds for 1 minutes and exit happily.